### PR TITLE
Various fixes and improvements

### PR DESCRIPTION
**Pull request by rdumusc**

### CODE REVIEW

File: tide/core/network/MPICommunicator.cpp [hunk 1]
```diff
@@ -44,6 +44,11 @@
 
 #include "utils/log.h"
 
+// WAR some deadlocks receiving MPI_IBcast with OpenMPI (version 1.10.2)
```
Comment (karjonas) [inline tide/core/network/MPICommunicator.cpp]: What does WAR mean in this context?

File: tide/core/utils/stl.h [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+                        std::inserter(diff, diff.begin()));
+    return diff;
+}
+
+/**
+ * Optimal implementation of bool contains(container, value).
+ * Uses container.find() if available, else uses generic std::find().
+ */
+template <class C, typename T>
+inline auto _contains_impl(const C& c, const T& v, int)
```
Comment (karjonas) [inline tide/core/utils/stl.h]: Why do you need int and long as the third argument?

File: tide/core/network/MPICommunicator.cpp (hunk 1 above)
Comment (rdumusc) [reply tide/core/network/MPICommunicator.cpp]: Haha yeah it stands for WorkARound, it's an abbreviation that was used in multiple places by Daniel and Stefan so I adopted it, despite I also didn't get it at first

File: tide/core/utils/stl.h (hunk 2 above)
Comment (rdumusc) [reply tide/core/utils/stl.h]: I assume it is needed for SFINAE to choose the container.find() variant when it's available, but I haven't checked in detail myself to be honest. I copied a solution provided here: https://codereview.stackexchange.com/questions/59997/contains-algorithm-for-stdvector

Comment (karjonas) [reply tide/core/utils/stl.h]: Yes, otherwise there would be a conflict by the two definitions.

File: tide/core/network/MPICommunicator.cpp (hunk 1 above)
Comment (karjonas) [reply tide/core/network/MPICommunicator.cpp]: If you want then change it to workaround.

**Pull request merged**